python-package/src/promptguard/promptguard_service.py

```diff
@@ -2,16 +2,22 @@
 This module exposes wrappers around API calls to the PromptGuard service.
 """
 import json
+import threading
 from dataclasses import dataclass
 from http import HTTPStatus
 from http.client import HTTPException
-from typing import Dict, List, Union
+from typing import Dict, List, Optional, Union
 
-from atls import AttestedHTTPSConnection, AttestedTLSContext
-from atls.validators import AZ_AAS_GLOBAL_JKUS, AzAasAciValidator, Validator
+import requests
+from atls.utils.requests import HTTPAAdapter
+from atls.validators import Validator
+from atls.validators.azure.aas import PUBLIC_JKUS, AciValidator
 from promptguard.authentication import get_api_key
 from promptguard.configuration import get_server_config
 
+_session: Optional[requests.Session] = None
+_sessionLock: threading.Lock = threading.Lock()
```

Comment:
Nit: `_session_lock` to match Python naming convention

Some comments about what these variables are used for would also help

Reply:
Added.